app/controllers/my-projects/assignment/recommendations/add.js

```diff
@@ -101,19 +101,38 @@ export default class MyProjectsProjectRecommendationsAddController extends Contr
 
   minDate = MINIMUM_VOTE_DATE;
 
-  // currently only used to name the universal queue
-  queueName = 'recommendation';
-
   isSubmitting = false;
 
   submitError = false;
 
+  @computed('assignment')
+  get assignmentQueueName() {
+    const parsedAssignmentId = this.assignment.id.replace(/-/g, '');
+    return `assign${parsedAssignmentId}`;
+  }
+
+  // This is the assignment-specific file queue.
+  // It holds all files related to this assignment, across all disposition queues.
+  // This queue is also different from the universal queue (global to app).
+  // Files are first uploaded to this queue. Then, upon submit, copied over to
+  // disposition queues.
+  // This queue is also used to list files next to the "upload" button.
+  // TODOS:
+  //    - Feature to remove files from this queue.
+  //    - When implementing "by action" file upload, only copy files from this queue
+  //      to dispositions queues if allActions === true.
+  @computed('fileQueue', 'assignmentQueueName')
+  get assignmentQueue() {
+    return this.fileQueue.find(this.assignmentQueueName) || this.fileQueue.create(this.assignmentQueueName);
+  }
```

Comment:
Nice splitting things out a bit more... I am worried about this though, because it breaks the declarative nature of computed properties through the use of that _conditionally_-invoked `create` method. A computed property that mutates some other data elsewhere introduces side-effects, makes stuff hard to debug. I'm gonna see if I can find another approach.. Thanks!